### PR TITLE
Comparing Auth changes

### PR DESCRIPTION
**Pull request by jplafonta**

### CODE REVIEW

File: code/include/playfab/PlayFabAuth.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/// Both the duplicated handle and the original handle need to be closed with PlayFabAuthenticatedEntityCloseHandle when they
+/// are no longer needed.
+/// </remarks>
+HRESULT PlayFabAuthContextDuplicateHandle(
+    _In_ PlayFabAuthContextHandle authHandle,
+    _Out_ PlayFabAuthContextHandle* duplicatedAuthHandle
+) noexcept;
+
+/// <summary>
+/// Closes a PlayFabAuthenticatedEntityHandle.
```
Comment (rakhad) [inline code/include/playfab/PlayFabAuth.h]: >PlayFabAuthenticatedEntityHandle [](start = 13, length = 32)

PlayFabAuthContextHandle?
 #Resolved

File: code/include/playfab/PlayFabAuth.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+/// If the auth call fails, authHandle with be null. Otherwise, the handle must be closed with PlayFabAuthenticatedEntityCloseHandle
+/// when it is no longer needed.
+/// </remarks>
+HRESULT PlayFabAuthGetAuthResult(
+    _In_ XAsyncBlock* async,
+    _Out_ PlayFabAuthContextHandle* authHandle
+) noexcept;
+
+/// <summary>
+/// Duplicates a PlayFabAuthenticatedEntityHandle.
```
Comment (rakhad) [inline code/include/playfab/PlayFabAuth.h]: >PlayFabAuthenticatedEntityHandle [](start = 17, length = 32)

PlayFabAuthContextHandle? #Resolved

File: code/include/playfab/PlayFabClientAuthApi.h [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+#pragma once
+
+#include <playfab/PlayFabAuth.h>
+#include <playfab/PlayFabClientDataModels_c.h>
+
+extern "C"
+{
+
+/// <summary>
+/// <LoginWithAndroidDeviceID documentation goes here.>
```
Comment (rakhad) [inline code/include/playfab/PlayFabClientAuthApi.h:17]: ><LoginWithAndroidDeviceID documentation goes here.> [](start = 4, length = 51)

Just to make sure - this is a task for a later pr? #WontFix

File: code/include/playfab/PlayFabClientAuthApi.h [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+/// <param name="stateHandle">PlayFabStateHandle returned from PlayFabInitialize call.</param>
+/// <param name="request">Populated request object.</param>
+/// <param name="async">XAsyncBlock for the async operation.</param>
+/// <returns>Result code for this API operation.</returns>
+/// <remarks>
+/// If successful, call <see cref="PlayFabClientGetAuthResult"/> to get the result.
+/// </remarks>
+HRESULT PlayFabClientLoginWithAndroidDeviceID(
+    _In_ PlayFabStateHandle stateHandle,
+    _In_ const PlayFabClientLoginWithAndroidDeviceIDRequest* request,
```
Comment (rakhad) [inline code/include/playfab/PlayFabClientAuthApi.h:28]: >PlayFabClientLoginWithAndroidDeviceIDRequest [](start = 15, length = 44)

Where are these defined? #Resolved

File: code/source/Client/ClientAuthApi.cpp [hunk 5]
```diff
@@ -0,0 +1,731 @@
...
+
+    m_httpClient.MakePostRequest(
+        "/Client/LoginWithAndroidDeviceID",
+        headers,
+        JsonUtils::ToJson(request),
+        queue,
+        callComplete
+    );
+}
+void ClientAuthAPI::LoginWithApple(
```
Comment (rakhad) [inline code/source/Client/ClientAuthApi.cpp:50]: nit: line break between  #WontFix

File: code/source/Client/ClientAuthApi.cpp [hunk 6]
```diff
@@ -0,0 +1,731 @@
...
+        }
+    };
+
+    m_httpClient.MakePostRequest(
+        "/Client/LoginWithApple",
+        headers,
+        JsonUtils::ToJson(request),
+        queue,
+        callComplete
+    );
```
Comment (rakhad) [inline code/source/Client/ClientAuthApi.cpp:82]: Can't this live as a helper method that takes in a http url, the request json, and the callbacks?

File: code/source/Server/ServerAuthApi.cpp [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        }
+    };
+
+    m_httpClient.MakePostRequest(
+        "/Server/LoginWithServerCustomId",
+        headers,
+        JsonUtils::ToJson(request),
+        queue,
+        callComplete
+    );
```
Comment (rakhad) [inline code/source/Server/ServerAuthApi.cpp:48]: Same comment as ClientAuthApi.cpp - could this be a helper function?

**Review by rakhad — APPROVED** [review]
:shipit:

File: code/include/playfab/PlayFabClientAuthApi.h (hunk 3 above)
Comment (jplafonta) [reply code/include/playfab/PlayFabClientAuthApi.h:17]: Ya. Will open a task so this doesn't get lost.  Not 100% sure how we will provide these docs but I have a couple ideas.

---
In reply to: [607288277](https://github.com/jplafonta/XPlatCSdk/pull/2#discussion_r607288277) [](ancestors = 607288277)

File: code/include/playfab/PlayFabClientAuthApi.h (hunk 4 above)
Comment (jplafonta) [reply code/include/playfab/PlayFabClientAuthApi.h:28]: PlayFabClientDataModels_c.h

---
In reply to: [607290987](https://github.com/jplafonta/XPlatCSdk/pull/2#discussion_r607290987) [](ancestors = 607290987)

File: code/source/Client/ClientAuthApi.cpp (hunk 6 above)
Comment (jplafonta) [reply code/source/Client/ClientAuthApi.cpp:82]: Which part?  Not sure what creating a helper method actually accomplishes here.

---
In reply to: [607292144](https://github.com/jplafonta/XPlatCSdk/pull/2#discussion_r607292144) [](ancestors = 607292144)

File: code/source/Client/ClientAuthApi.cpp (hunk 5 above)
Comment (jplafonta) [reply code/source/Client/ClientAuthApi.cpp:50]: added in another branch that I will PR soon.  Going to leave it as is now so it doesn't conflict

---
In reply to: [607291536](https://github.com/jplafonta/XPlatCSdk/pull/2#discussion_r607291536) [](ancestors = 607291536)

**Pull request merged**